starter_code/src/movies.js

```diff
@@ -1,15 +1,115 @@
 /* eslint no-restricted-globals: 'off' */
 
 // Iteration 1: Ordering by year - Order by year, ascending (in growing order)
+function orderByYear(movies) {
+
+  // Shallow copy
+  let moviesYears = movies.map( (movie, i) => { return movie });
+
+  // Sort movies by year
+  moviesYears.sort( (a,b) => {
+      return a.year - b.year;
+  });
+
+  // Sort movies of the same year by title
+  moviesYears.sort( (a,b) => {
+    if (a.year === b.year) {
+      return a.title === b.title ? 0 : a.title > b.title ? 1 : -1;
+    }
+  });
+
+  return moviesYears;
+}
+
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct
+function howManyMovies(movies) {
+
+    let filteredMovies = movies.filter((movie,i) => {
+        return movie.director === "Steven Spielberg" && movie.genre.includes("Drama");
+    });
+
+    return filteredMovies.length;
+}
 
 // Iteration 3: Alphabetic Order - Order by title and print the first 20 titles
+function orderAlphabetically(movies) {
+
+  // Shallow copy - create an Array of movie titles
+  let moviesList = movies.map( (movie, i) => {
+    return movie.title;
+  });
+
+  // Order the list alphabetically
+  moviesList.sort( (a,b) => {
+    return a === b ? 0 : a > b ? 1 : -1;
+  });
+
+  // Return first 20 elements
+  return moviesList.slice(0,20);
+}
```

Comment:
Here again you could chain everything at once

```
function orderAlphabetically(movies) {
  return movies.map(m => m.title).sort((a,b) => a.localeCompare(b)).slice(0,20)
}
```

The map creates a new array, then your sort it, then you slice it (which returns a new array)

Reply:
Agreed ! This one was very verbose for no reason.